File: tests/parsers/fixtures/sample.rb
# Sample Ruby module for parser testing.

require "json"
require_relative "helpers"

module Sample
  # Processes data records.
  class DataProcessor
    attr_reader :config

    def initialize(config)
      @config = config
    end

    def process(data)
      data.map { |item| transform(item) }
    end

    private

    def transform(item)
      item.strip.downcase
    end
  end
end

File: tests/parsers/fixtures/sample.py
"""Sample Python module for parser testing."""

import os
from pathlib import Path


class DataProcessor:
    """Processes data records."""

    def __init__(self, config: dict) -> None:
        self.config = config

    def process(self, data: list[str]) -> list[str]:
        """Process a list of data strings."""
        return [self._transform(item) for item in data]

    def _transform(self, item: str) -> str:
        result = item.strip().lower()
        return result


def load_data(filepath: str) -> list[str]:
    """Load data from a file."""
    path = Path(filepath)
    return path.read_text().splitlines()
